app/services/update_work_package_service.rb

```diff
@@ -42,10 +42,21 @@ def update
     work_package.update_by!(user, effective_params)
   end
 
+  ##
+  # Save changes to the work package itself.
+  # Calls save_journals implicitly as a save hook
   def save
     work_package.save
   end
 
+  ##
+  # Create journal for the given user and note.
+  # Does not change the work package itself.
+  def add_journal(notes = '')
```

Comment:
How about renaming this to `create_journal` to communicate the saving within the method.